### DISCUSSION TITLE
How better fix stale row data with NTIE table refresh

### DISCUSSION THREAD

**Discussion by podliashanyk** [Uninett/Howitz#97] [Ideas]
Since table refresh with NTIE in #87 is targeted, only updated rows are replaced leading to some info in the columns becoming stale for the rows that haven't been updated for a while, f.e. "AGE" and "DOWNTIME":
    <img width="240" alt="Screenshot 2024-05-28 at 16 16 20" src="https://github.com/Uninett/Howitz/assets/60876078/ad4c31f6-ff0e-4496-9bde-fe088ce4b288">

Should we implement a complete table refresh (the old way) every X **minutes** in Howitz? Or should we extend `UpdateHandler` in zinolub? Any other ideas on how to better solve this? Is the time ripe for server-sent events?

**Comment by hmpf** [comment]
It certainly does not belong in UpdateHandler, that's only for ntie-stuff.

**Comment by podliashanyk** [comment]
Server-sent events also seem like overkill for this single issue to me. 
The remaining alternatives I can think of are:
- "old-style" refresh of whole table every 2 minutes or so
- or each row requesting refresh of its time-related data ("age", "downtime", "lasttrans" etc) every minute or so. This would likely require a new endpoint in zinolib.
- optimistic time-related data refresh of each row where no request it sent to zinolib. Howitz just assumes if time-related value on event (say "downtime) wasn't updated by `UpdateHandler` then a normal timeflow is assumed for it therefore it can safely be increased by expected timedelta.

**Reply by hmpf** [reply]
If "old-style" starts at the top and works its way down that might work. Less annoying than the last number in the entire column suddenly changing.

I think faking it (third option) might be best.

We're lucky we don't show seconds =)

**Reply by podliashanyk** [reply]
> If "old-style" starts at the top and works its way down that might work. Less annoying than the last number in the entire column suddenly changing.

Old style reswaps all of the rows at once.

> I think faking it (third option) might be best.

Yes, sounds most efficient. 

> We're lucky we don't show seconds =)

Oh yeah!

**Comment by hmpf** [comment]
I'm looking at curitz, it does not update the entire Age-column every minute. Only some of them and it looks a bit weird frankly. I don't understand **why** something updates and there seems to be groups of rows updating at the same time.

Maybe we should ask users. I would probably not like the entire column updating every minute because I suspect it would be uncomfortable to watch. Like very.. slow.. flashing..

**Reply by podliashanyk** [reply]
> Maybe we should ask users. I would probably not like the entire column updating every minute because I suspect it would be uncomfortable to watch. Like very.. slow.. flashing..

It shouldn't blink when it updates so should be fine I believe 😅 Those updates should be as "quiet" as NTIE row updates. But I guess the spinner indicator should be disabled completely for this kind of update. To make it as silent and smooth as possible. 
Some sort of update will be needed anyways. Like the idea to ask users!

**Reply by hmpf** [reply]
My eyes will notice it if an entire column of numbers updates at once I suspect. Sudden, scary change :)
